src/main/java/com/ericsson/ei/notifications/JenkinsCrumb.java

```diff
@@ -95,7 +95,9 @@ public JsonNode fetchJenkinsCrumb(String encoding, String url)
      */
     private URL buildJenkinsCrumbUrl(String url) throws MalformedURLException {
         String baseUrl = urlParser.extractBaseUrl(url);
-        URL jenkinsCrumbUrl = new URL(baseUrl + JENKINS_CRUMB_ENDPOINT);
+        String contextPath = urlParser.extractContextPath(url).split("/")[1];
+        String jenkinsBaseUrl = String.format("%s/%s", baseUrl, contextPath);
```

Comment:
Is this valid for valid for all Jenkins versions? Does this work with previous Jenkins versions as well?
What has changed between previous Jenkins version bad new Jenkins version for handling Crumb Auth?

Reply:
I checked the subscription,, I guess url is
http://<JenkinsHost:port>/job/<JobName>/job/<branch>/build
Context-path refers in current code base rest api entrypoint path to trigger the jenkins build.
So code need to rewritten so it might be clear.
And I guess you want handle the Tomcat Context-path that Tomcat adds in the middle.
That conext-path is more the real context path.
So code need to re-written a bit to in a smart way figure out the real context-path that is added by Tomcat in url address and in smart way extract where does the Jenkins url/entrypoint path starts.

Is this the case you try to solve?
http://<JenkinsHost:port>/<web server application context path>/job/<JobName>/job/<branch>/build

Reply:
I agree with Tobias here, this is a fix for an Ericsson specific problem that will break the functionality for everyone else.

Reply:
One suggestion to solve this is to introduce context-path tagging in the Jenkins url.
If we have this full url Jinkens jobb:
http://some_host:12345/jenkins-context-path/job/hello-word/build

In subscription Meta field user need to replace the slahses with some other character/tag, for marking that this is context-path, example if we can use '|' as tag/context-path marker:
http://some_host:12345|jenkins-context-path/job/hello-word/build

In EI code we can then check if url has "|" character , if it has that character in the first part of url after port-number, then its a context-path. Based on the we can devide it up to url variables in java code, and replace '|' with a '/' :
String jenkinsBaseUrl="http://some_host:12345/jenkins-context-path"
String jenkinsJobBuildUrlPath=/job/hello-word/build

Now you can make a CSRF Crumb authentication with help of "jenkinsBaseUrl" variable.
To trigger the jenkins job after CSRF Crumb authentication, then contactenate these to variables to full url and make the HTTP request for triggering the job.


I think this change is backward compatible and it will work with and without context-path.

Documentation in Frontend and in the Web-UI need to be made, so its clear to user how it works with context-paths Jenkins url address.

Reply:
Yes this approach sounds good, and we also assumed the same like but checking with the keyword /job after base url depending on this we can differentiate between with contextPath and without contextPath assuming the contextPath lies in between the baseUrl and /job.
But here we have few doubts like whether the context path can be "/jenk/ins" can it have / also , if this is the case we need to handle such.
So currently we are checking jenkins rules and url standards and we will try once in local the suggestion proposed.

Reply:
@Tobias can we proceed with with the above solution assuming the jenkins url will be:
http://some_host:12345/jenkins-context-path/job/hello-word/build

considering the contextpath lies between base url and /job

Reply:
Ok, I guess its an OK solution. I am no super expert on which Jenkins url entrypoints paths.
As long test goes successful, then its a OK solution.
I guess we have some Subscription Jenkins Triggering tests in System tests in EI Frontend git repository as well, those tests need to be executed as well, you can do that manually on your local computer or something.

Reply:
Ok Tobias , will test the chages